PanelViewController/Framework source/PanelViewController.swift

```diff
@@ -256,45 +256,46 @@ class PanelViewController: UIViewController {
             isFirstLayout = false
             paneView.frame = CGRect(x: 0, y: paneY, width: viewSize.width, height: (viewSize.height + 88) - paneY)
         }
+        
         //1. per specs, set backView frame
         backViewController?.view.frame = view.bounds
         //2. backViewOverlay frame has to be the same as backViewController, per specs.
         //   if backViewControllerFrame is nil, then there should not be an overlay.
         backViewOverlay.frame = backViewController?.view.frame ?? CGRect.zero
 
-        let offset: CGFloat = floatingHeaderHeight
         //To hide the corner radius curve at the bottom increase the height of the dragHandleView
         let bottomPadding: CGFloat = 10
-        dragHandleView.frame = CGRect(x: 0, y: offset, width: paneView.bounds.width, height: closedHeight + bottomPadding)
+        dragHandleView.frame = CGRect(x: 0, y: floatingHeaderHeight, width: paneView.width, height: closedHeight + bottomPadding)
         
         if let floatingHeaderView = floatingHeaderView {
-            let floatingHeaderMinY = self.floatingHeaderMinY ?? view.bounds.height / 2
-            let frame = CGRect(x: 0, y: 0, width: paneView.bounds.width, height: floatingHeaderHeight)
-            if paneView.frame.origin.y < floatingHeaderMinY {
-                let floatOffset = floatingHeaderMinY - paneView.frame.origin.y
-                floatingHeaderView.frame = CGRect(x: frame.origin.x, y: floatOffset, width: frame.size.width, height: floatingHeaderHeight)
+            let floatingHeaderMinY = self.floatingHeaderMinY ?? view.height / 2
+            if paneView.y < floatingHeaderMinY {
+                floatingHeaderView.y = floatingHeaderMinY - paneView.y
             } else {
-                floatingHeaderView.frame = CGRect(x: frame.origin.x, y: frame.origin.y, width: frame.size.width, height: floatingHeaderHeight)
+                floatingHeaderView.y = 0
```

Comment:
Question about your extension.  If Apple's way is to not be able to set the individual attributes like height, origin y, etc, except by setting the entire frame, are we subverting their intention?

Reply:
I don't think so. Under the hood, all the extension is doing is setting the frame. The extension is just giving shortcuts to change these attributes, in order to cut down on boilerplate code.

Of course, Apple hasn't provided these shortcuts for us, but then UIKit is full of little weirdnesses like that: For instance, you can't just say `button.title = "OK"`; you have to use `button.setTitle:forState:`. And, you can't initialize a UIColor with a hex string.

For me, the bottom line is that Apple provided us with a mechanism to expand the language and environment, and by doing that, we aren't subverting anything; rather, we are taking advantage of the tools they gave us to improve our code.